src/panels/lovelace/cards/hui-calendar-card.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  css,
+  CSSResult,
+  customElement,
+  html,
+  LitElement,
+  property,
+  PropertyValues,
+  TemplateResult,
+} from "lit-element";
+
+import "../../../components/ha-card";
+import "../../../components/ha-icon";
+import "../../calendar/ha-full-calendar";
+
+import type {
+  HomeAssistant,
+  CalendarEvent,
+  Calendar,
+  CalendarViewChanged,
+} from "../../../types";
+import type { LovelaceCard } from "../types";
+import type { CalendarCardConfig } from "./types";
+import { findEntities } from "../common/find-entites";
+import { applyThemesOnElement } from "../../../common/dom/apply_themes_on_element";
+import "../components/hui-warning";
+import { processConfigEntities } from "../common/process-config-entities";
+import { computeStateName } from "../../../common/entity/compute_state_name";
+import { fetchCalendarEvents } from "../../../data/calendar";
+import { HASSDomEvent } from "../../../common/dom/fire_event";
+import { HA_COLOR_PALETTE } from "../../../common/const";
+
+@customElement("hui-calendar-card")
+export class HuiCalendarCard extends LitElement implements LovelaceCard {
+  // public static async getConfigElement(): Promise<LovelaceCardEditor> {
+  //   await import(
+  //     /* webpackChunkName: "hui-calendar-card-editor" */ "../editor/config-elements/hui-calendar-card-editor"
+  //   );
+  //   return document.createElement("hui-calendar-card-editor");
+  // }
+
+  public static getStubConfig(
+    hass: HomeAssistant,
+    entities: string[],
+    entitiesFill: string[]
+  ) {
+    const includeDomains = ["calendar"];
+    const maxEntities = 1;
+    const foundEntities = findEntities(
+      hass,
+      maxEntities,
+      entities,
+      entitiesFill,
+      includeDomains
+    );
+
+    return {
+      entity: foundEntities[0] || "",
+    };
+  }
+
+  @property() public hass?: HomeAssistant;
+
+  @property() private _config?: CalendarCardConfig;
+
+  @property() public _events: CalendarEvent[] = [];
+
+  @property() private _calendars: Calendar[] = [];
+
+  public setConfig(config: CalendarCardConfig): void {
+    if (!config.entities) {
+      throw new Error("Entities must be defined");
+    }
+
+    if (config.entities && !Array.isArray(config.entities)) {
+      throw new Error("Entities need to be an array");
+    }
+
+    this._config = config;
+  }
+
+  public getCardSize(): number {
+    return 4;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass || !this._calendars.length) {
+      return html``;
+    }
+
+    return html`
+      <ha-card .header=${this._config.title}>
+        <ha-full-calendar
+          .events=${this._events}
+          .narrow=${true}
+          .hass=${this.hass}
+          .views=${["listWeek", "dayGridMonth"]}
```

Comment:
This tells me `string[] is not assignable to FullCalendarView[]`

Reply:
You can ignore it. Or try `["listWeek", "dayGridMonth"] as FullCalendarView[]`